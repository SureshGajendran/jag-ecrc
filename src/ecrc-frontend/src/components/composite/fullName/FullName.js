import React from "react";
import PropTypes from "prop-types";

import { SimpleForm } from "../simpleForm/SimpleForm";

export default function FullName({
  title,
  fullname: { legalFirstNm, legalSecondNm, legalSurnameNm }
}) {
  const name = {
    title,
    textInputs: [legalFirstNm, legalSecondNm, legalSurnameNm],
    buttons: []
  };

  return <SimpleForm simpleForm={name} />;
}

FullName.propTypes = {
  title: PropTypes.string,
  fullname: PropTypes.shape({
    legalFirstNm: PropTypes.shape({
      label: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      value: PropTypes.string,
      textInputStyle: PropTypes.string
    }),
    legalSecondNm: PropTypes.shape({
      label: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      value: PropTypes.string,
      textInputStyle: PropTypes.string
    }),
    legalSurnameNm: PropTypes.shape({
      label: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      value: PropTypes.string,
      textInputStyle: PropTypes.string
    })
  })
};

FullName.defaultProps = {
  title: "",
  fullname: {
    firstName: {
      value: "",
      textInputStyle: ""
    },
    middleName: {
      value: "",
      textInputStyle: ""
    },
    lastName: {
      value: "",
      textInputStyle: ""
    }
  }
};
