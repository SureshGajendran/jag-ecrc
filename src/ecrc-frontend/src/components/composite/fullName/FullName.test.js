import React from "react";
import { create } from "react-test-renderer";

import FullName from "./FullName";

describe("FullName Component", () => {
  test("Matches the snapshot", () => {
    const fullname = {
      legalFirstNm: {
        label: "First Name",
        id: "firstName",
        value: "First",
        textInputStyle: "textinput_non_editable_gray"
      },
      legalSecondNm: {
        label: "Middle Name",
        id: "middleName",
        value: "Middle",
        textInputStyle: "textinput_non_editable_gray"
      },
      legalSurnameNm: {
        label: "Last Name",
        id: "lastName",
        value: "Last",
        textInputStyle: "textinput_non_editable_gray"
      }
    };

    const testFullName = create(
      <FullName title={"Full Name"} fullname={fullname} />
    );
    expect(testFullName.toJSON()).toMatchSnapshot();
  });
});
