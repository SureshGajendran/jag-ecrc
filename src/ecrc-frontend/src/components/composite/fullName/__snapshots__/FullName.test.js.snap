// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullName Component Matches the snapshot 1`] = `
<form
  className="simpleForm"
>
  <div
    className="simpleForm_title"
  >
    Full Name
  </div>
  <ul>
    <li>
      <div>
        <div
          className="label"
        >
          <label
            className="textinput_label"
            htmlFor="firstName"
          >
            First Name
          </label>
          
           
          <span
            className="note"
          />
        </div>
        <input
          className="textinput_non_editable_gray"
          defaultValue="First"
          id="firstName"
          onChange={[Function]}
          readOnly={true}
          type="text"
        />
        <span
          className="error"
        />
      </div>
    </li>
    <li>
      <div>
        <div
          className="label"
        >
          <label
            className="textinput_label"
            htmlFor="middleName"
          >
            Middle Name
          </label>
          
           
          <span
            className="note"
          />
        </div>
        <input
          className="textinput_non_editable_gray"
          defaultValue="Middle"
          id="middleName"
          onChange={[Function]}
          readOnly={true}
          type="text"
        />
        <span
          className="error"
        />
      </div>
    </li>
    <li>
      <div>
        <div
          className="label"
        >
          <label
            className="textinput_label"
            htmlFor="lastName"
          >
            Last Name
          </label>
          
           
          <span
            className="note"
          />
        </div>
        <input
          className="textinput_non_editable_gray"
          defaultValue="Last"
          id="lastName"
          onChange={[Function]}
          readOnly={true}
          type="text"
        />
        <span
          className="error"
        />
      </div>
    </li>
  </ul>
</form>
`;
